concept/validation/schema-validation.feature

```diff
@@ -503,243 +747,332 @@ Feature: Schema validation
-      rel01 sub relation, relates role01;
-      ent0 sub entity, plays rel00:role00, plays rel01:role01;
-
-      rule make-me-illegal:
-      when {
-        (role00: $e) isa rel00;
-      } then {
-        (role01: $e) isa rel01;
-      };
-    """
-    Given transaction commits
-
-    When session opens transaction of type: write
-    Then relation(rel00) set label: renamed-rel00
-    Then transaction commits; throws exception
-
-
-  Scenario: Types which are referenced in rules may not be deleted
-    Given typeql define
-    """
-    define
-      rel00 sub relation, relates role00, relates extra_role;
-      rel01 sub relation, relates role01;
-      rel1 sub rel00;
-      ent0 sub entity, plays rel00:role00, plays rel01:role01;
-
-      rule make-me-illegal:
-      when {
-        (role00: $e) isa rel1;
-      } then {
-        (role01: $e) isa rel01;
-      };
-    """
-    Given transaction commits
-
-
-    When session opens transaction of type: write
-    Then delete relation type: rel01; throws exception
-
-    When session opens transaction of type: write
-    Then delete relation type: rel1; throws exception
-
-    When session opens transaction of type: write
-    Then relation(rel01) unset related role: role01; throws exception
-
-    # We currently can't do this at operation time, so we check at commit-time
-    When session opens transaction of type: write
-    Then relation(rel00) unset related role: role00
-    Then transaction commits; throws exception
-
-
-  Scenario: Rules made unsatisfiable by schema modifications are flagged at commit time
-    Given typeql define
-    """
-    define
-      rel00 sub relation, relates role00, relates extra_role;
-      rel01 sub relation, relates role01;
-      rel1 sub rel00;
-
-      ent00 sub entity, abstract, plays rel00:role00, plays rel01:role01;
-      ent01 sub entity, abstract;
-      ent1 sub ent00;
-
-      rule make-me-unsatisfiable:
-      when {
-        $e isa ent1;
-        (role00: $e) isa rel1;
-      } then {
-        (role01: $e) isa rel01;
-      };
-    """
-    Given transaction commits
-
-    When session opens transaction of type: write
-    Then relation(rel1) set supertype: rel01
-    Then transaction commits; throws exception
-
-    When session opens transaction of type: write
-    Then entity(ent00) unset plays: rel00:role00
-    Then transaction commits; throws exception
-
-    When session opens transaction of type: write
-    Then entity(ent1) set supertype: ent01
-    Then transaction commits; throws exception
-
-
-  Scenario: Annotations on ownership overrides must be atleast as strict as the overridden ownerships
-    Given create attribute type: attr0, with value type: string
-    Given attribute(attr0) set abstract: true
-    Given create attribute type: attr1, with value type: string
+    When connection open schema transaction for database: typedb
+    When entity(ent1) set supertype: ent01; fails
+
+#  TODO: Refactor to functions
```

Comment:
Gonna rename this file to `functions` after I move everything else and leave it until we refactor these steps.